lib/ansible/modules/network/routeros/routeros_facts.py

```diff
@@ -0,0 +1,432 @@
+
+    def parse_model(self, data):
+        match = re.search(r'model:\s(.*)\s*$', data, re.M)
+        if match:
+            return match.group(1)
+
+    def parse_serialnum(self, data):
+        match = re.search(r'serial-number:\s(.*)\s*$', data, re.M)
+        if match:
+            return match.group(1)
+
+
+class Hardware(FactsBase):
+
+    COMMANDS = [
+        '/system resource print without-paging'
+    ]
+
+    def populate(self):
+        super(Hardware, self).populate()
+        data = self.responses[0]
+        if data:
+            self.parse_filesystem_info(data)
+            self.parse_memory_info(data)
+
+    def parse_filesystem_info(self, data):
+        match = re.search(r'free-hdd-space:\s(.*)([KMG]iB)', data, re.M)
+        if match:
+            self.facts['spacefree_mb'] = self.to_megabytes(match)
+        match = re.search(r'total-hdd-space:\s(.*)([KMG]iB)', data, re.M)
+        if match:
+            self.facts['spacetotal_mb'] = self.to_megabytes(match)
+
+    def parse_memory_info(self, data):
+        match = re.search(r'free-memory:\s(\d+\.?\d*)([KMG]iB)', data, re.M)
+        if match:
+            self.facts['memfree_mb'] = self.to_megabytes(match)
+        match = re.search(r'total-memory:\s(\d+\.?\d*)([KMG]iB)', data, re.M)
+        if match:
+            self.facts['memtotal_mb'] = self.to_megabytes(match)
+
+    def to_megabytes(self, data):
+        if data.group(2) == 'KiB':
+            return float(data.group(1)) / 1024
+        elif data.group(2) == 'MiB':
+            return float(data.group(1))
+        elif data.group(2) == 'GiB':
+            return float(data.group(1)) * 1024
+        else:
+            return None
+
+
+class Config(FactsBase):
+
+    COMMANDS = ['/export']
+
+    def populate(self):
+        super(Config, self).populate()
+        data = self.responses[0]
+        if data:
+            self.facts['config'] = data
+
+
+class Interfaces(FactsBase):
+
+    COMMANDS = [
+        '/interface print detail without-paging',
+        '/ip address print detail without-paging',
+        '/ipv6 address print detail without-paging',
+        '/ip neighbor print detail without-paging'
+    ]
+
+    def populate(self):
+        super(Interfaces, self).populate()
+
+        self.facts['interfaces'] = dict()
+        self.facts['all_ipv4_addresses'] = list()
+        self.facts['all_ipv6_addresses'] = list()
+
+        data = self.responses[0]
+        if data:
+            interfaces = self.parse_interfaces(data)
+            self.populate_interfaces(interfaces)
+
+        data = self.responses[1]
+        if data:
+            data = self.parse_addresses(data)
+            self.populate_ipv4_interfaces(data)
+
+        data = self.responses[2]
+        if data:
+            data = self.parse_addresses(data)
+            self.populate_ipv6_interfaces(data)
+
+        data = self.responses[3]
+        if data:
+            self.facts['neighbors'] = self.parse_neighbors(data)
```

Comment:
This causes `neighbors` to not be in the results if none are detected, contradicting the documentation of the `ansible_net_neighbors` return value.

Reply:
Thank you. I will add an empty dict as a placeholder for `neighbors`.